docs/architecture.rst

```diff
@@ -122,22 +128,58 @@ strategy and the use case.
 
 Data in offline segments are immmutable (Rows cannot be added, deleted, or modified). However, segments may be replaced with modified data.
 
+.. _ingesting-realtime-data:
+
 Ingesting Realtime Data
 ^^^^^^^^^^^^^^^^^^^^^^^
-Segments for realtime tables are constructed by Pinot servers. The servers ingest rows from realtime streams (such as
-Kafka) until
-some completion threshold (such as number of rows, or a time threshold) and build a segment out of those rows. Depending
-on the type of ingestion mechanism used (stream or partition level), segments may be locally stored in the servers
-or in the controller's segment store.
+Segments for realtime tables are constructed by Pinot servers with rows ingested from data streams such as Kafka.
+Rows ingested from streams are made available for query processing as soon as they are ingested, thus enabling
+applications such as those that need real-time charts on analytics.
+
+In large scale installations, data in streams is typically split across multiple stream partitions. The underlying
+stream may provide consumer implementations that allow applications to consume data from any subset of partitions,
+including all partitions (or, just from one partition).
+
+A pinot table can be configured to consume from streams in one of two modes:
+
+    * ``LowLevel``: This is the preferred mode of consumption. Pinot creates independent partition-level consumers for
```

Comment:
Dint we rename them to partition level and stream level.

Reply:
no